ably/proto/message.go

```diff
@@ -22,12 +27,223 @@ type Message struct {
 	ClientID     string                 `json:"clientId,omitempty" codec:"clientId,omitempty"`
 	ConnectionID string                 `json:"connectionId,omitempty" codec:"connectionID,omitempty"`
 	Name         string                 `json:"name,omitempty" codec:"name,omitempty"`
-	Data         string                 `json:"data,omitempty" codec:"data,omitempty"`
+	Data         *DataValue             `json:"data,omitempty" codec:"data,omitempty"`
```

Comment:
What's the reasoning behind having a separate `DataValue` type, instead of having this functionality in `Message`?